feat: Type imports #GUILD-349

yoo-brh opened the pull request.

yoo-brh committed:
feat: Type imports #GUILD-349

kelvinlouis commented on packages/eslint-config-base/index.js:
```diff
@@ -59,6 +59,13 @@ module.exports = {
         allowTypedFunctionExpressions: true,
       },
     ],
+    '@typescript-eslint/consistent-type-imports': [
```
what happens when eslint fixes the issue?
will it inline the types when there is already an import statements for non-types, or will it create a separate type import?

i do not see in the documentation what happens if no fixStyle is provided. my guess it uses the default: https://typescript-eslint.io/rules/consistent-type-imports/

yoo-brh replied:
I added `fixStyle: 'inline-type-imports'` by default but the rule `'import/consistent-type-specifier-style': 'off'` is there to don't raise an error if we use separate type imports e.g.: 
`import type { ChangeEventHandler, FocusEvent, SyntheticEvent } from 'react';`

yoo-brh committed:
Added default fixStyle #GUILD-349

RadLikeWhoa commented:
LGTM. Might want to bump the version as well.

yoo-brh committed:
updated lerna

yoo-brh committed:
Publish

 - @yoo-digital/eslint-config-angular@18.0.0
 - @yoo-digital/eslint-config-base@3.0.0-canary.0
 - @yoo-digital/eslint-config-react@3.0.0-canary.0
